api/server/cluster_test.go

```diff
@@ -572,6 +573,252 @@ func TestGetNodeIdFromIpFailed(t *testing.T) {
+	defer tc.Finish()
+
+	restClient, err := clusterclient.NewClusterClient(ts.URL, "v1")
+	assert.NoError(t, err)
+
+	// Set expections
+	tc.MockCluster().
+		EXPECT().
+		NodeStatus().
+		Return(api.Status_STATUS_NONE, fmt.Errorf("error in node status")).
+		Times(1)
+
+	// Check status
+	status, err := clusterclient.ClusterManager(restClient).NodeStatus()
+	assert.Error(t, err)
+	assert.Equal(t, api.Status_STATUS_NONE, status)
+	assert.Contains(t, err.Error(), "error in node status")
+}
+
+func TestEnumerateAlertsFailed(t *testing.T) {
+
+	// Create a new global test cluster
+	ts, tc := testClusterServer(t)
+	defer ts.Close()
+	defer tc.Finish()
+
+	// time frame is exactly 24 hrs from current time.
+	endTime := time.Now()
+	startTime := endTime.Add(-24 * time.Hour)
+
+	// mock the cluster response
+	tc.MockCluster().
+		EXPECT().
+		EnumerateAlerts(gomock.Any(), gomock.Any(), gomock.Any()).
+		Return(&api.Alerts{}, fmt.Errorf("error in enumerate alerts"))
+
+	// create a cluster client to make the REST call
+	c, err := clusterclient.NewClusterClient(ts.URL, "v1")
+	assert.NoError(t, err)
+
+	// make the REST call
+	restClient := clusterclient.ClusterManager(c)
+	resp, err := restClient.EnumerateAlerts(startTime, endTime, api.ResourceType_RESOURCE_TYPE_NODE)
+
+	assert.Error(t, err)
+	assert.Nil(t, resp)
+
+	assert.Contains(t, err.Error(), "error in enumerate alerts")
+}
+
+func TestGetClusterConfFailed(t *testing.T) {
+
+	// Create a new global test cluster
+	ts, tc := testClusterServer(t)
+	defer ts.Close()
+	defer tc.Finish()
+
+	// mock the cluster response
+	tc.MockCluster().
+		EXPECT().
+		GetClusterConf().
+		Return(nil, fmt.Errorf("error in getting cluster config"))
+
+	// create a cluster client to make the REST call
+	c, err := clusterclient.NewClusterClient(ts.URL, "v1")
+	assert.NoError(t, err)
+
+	// make the REST call
+	restClient := clusterclient.ClusterManager(c)
+	resp, err := restClient.GetClusterConf()
+	assert.Error(t, err)
+	assert.Nil(t, resp)
+	assert.Contains(t, err.Error(), "error in getting cluster config")
+}
+
+func TestPeerStatusFailed(t *testing.T) {
+
+	// Create a new global test cluster
+	ts, tc := testClusterServer(t)
+	defer ts.Close()
+	defer tc.Finish()
+
+	// create a cluster client to make the REST call
+	c, err := clusterclient.NewClusterClient(ts.URL, "v1")
+	assert.NoError(t, err)
+
+	listenerName := "pxd"
+	// mock the cluster response
+	tc.MockCluster().
+		EXPECT().
+		PeerStatus(listenerName).
+		Return(nil, fmt.Errorf("error in peer status"))
+
+		// make the REST call
+	restClient := clusterclient.ClusterManager(c)
+
+	statusMap, err := restClient.PeerStatus(listenerName)
+	assert.Error(t, err)
+	assert.Nil(t, statusMap)
+	assert.Contains(t, err.Error(), "error in peer status")
+}
+
+func TestSetClusterConfFailed(t *testing.T) {
+
+	// Create a new global test cluster
+	ts, tc := testClusterServer(t)
+	defer ts.Close()
+	defer tc.Finish()
+
+	secretsConfig := &osdconfig.SecretsConfig{
```

Comment:
recommend to use new(type).Init() syntax available for osdconfig structs, but what you have here should work as well.

Reply:
Interesting, that is normally not idiomatic golang. The method I used is the one used normally. But both work.